CMakeLists.txt

```diff
@@ -0,0 +1,135 @@
+#
+# CMakeLists file for OpenSSL
+# Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+
+cmake_minimum_required(VERSION 3.14)
+
+project(OpenSSL LANGUAGES C CXX
+        DESCRIPTION "Open-source SSL and Cryptographic library"
+        HOMEPAGE_URL "https://github.com/openssl/openssl")
+
+# Configure the OpenSSL project
+if(OPENSSL_CONFIGURE_OPTIONS)
+  if (NOT PERL_EXECUTABLE) # Makes PERL_EXECUTABLE overridable
+    find_package(Perl 5.10.0 REQUIRED)
+  endif()
+  execute_process(
+    COMMAND ${CMAKE_COMMAND} -E env "BUILDFILE=openssl.cmake" ${PERL_EXECUTABLE} Configure ${OPENSSL_CONFIGURE_OPTIONS}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    RESULT_VARIABLE OPENSSL_CONFIGURE_RESULT
+    COMMAND_ECHO STDOUT
+  )
+  if(NOT (${OPENSSL_CONFIGURE_RESULT} EQUAL "0"))
+    message(FATAL_ERROR "Failed to configure OpenSSL with error: ${OPENSSL_CONFIGURE_RESULT}")
+  endif()
+endif()
+
+# Include the generated CMake config.
+include(openssl.cmake)
```

Comment:
Please check if file `openssl.cmake` exists before calling `include(openssl.cmake)`.
If file does not exist, please issue informative error message like "OpenSSL requires OPENSSL_CONFIGURE_OPTIONS variable set to be configured with CMake. Please read NOTES-CMAKE.md" or something like this.
Right now after calling FetchContent_MakeAvailable users get cryptic "Failed to find openssl.cmake" file.